iati/tests/test_version.py

```diff
@@ -384,105 +179,414 @@ def modifiable_attrib(self, request):
+        """Return the version parameter, but only if it's fully supported by pyIATI. Check undertaken with decorator."""
+        return version
+
+    @iati.version.allow_known_version
+    def return_known_version(version):  # pylint: disable=no-self-argument
+        """Return the version parameter, but only if it's known of by pyIATI. Check undertaken with decorator."""
+        return version
+
+    @iati.version.allow_possible_version
+    def return_possibly_version(version):  # pylint: disable=no-self-argument
+        """Return the version parameter, but only if it's known of by pyIATI. Check undertaken with decorator."""
+        return version
+
+    @pytest.fixture(params=[return_fully_supported_version])
+    def decorated_func_full_support(self, request):
+        """Return a decorated function that returns a version of the IATI Standard that is fully supported by pyIATI."""
+        return request.param
+
+    @pytest.fixture(params=[return_known_version])
+    def decorated_func_known(self, request):
+        """Return a decorated function that returns a version of the IATI Standard that pyIATI knows exists."""
+        return request.param
+
+    @pytest.fixture(params=[
+        return_possibly_version,
+        iati.version._prevent_non_version_representations
+    ])
+    def possibly_version_func(self, request):
+        """Return a function that returns a value that represents a possible IATI Version. Other values cause an error."""
+        return request.param
+
+    @pytest.fixture(params=[
+        iati.version._is_fully_supported,
+        iati.version._is_known
+    ])
+    def truthy_func(self, request):
+        """Return a function to check whether an input value is True or False based on whether it's a valid version."""
+        return request.param
+
+    @pytest.fixture(params=[
+        return_fully_supported_version,
+        return_known_version
+    ])
+    def decorated_func(self, request):
+        """Return a function to restrict whether an input value is a valid version, and raise a ValueError if it is not."""
+        return request.param
+
+    @pytest.fixture(params=[
+        return_fully_supported_version,
+        iati.version._is_fully_supported,
+        return_known_version,
+        iati.version._is_known
+    ])
+    def func_to_test(self, request):
+        """Return a function to check for TypeErrors being raised when provided values other than iati.Versions."""
+        return request.param
+
+
+class TestVersionSupportChecks(VersionSupportChecksBase):
+    """A container for tests relating to the detection of how much pyIATI supports particular versions."""
+
+    def test_fully_supported_version_fully_supported(self, std_ver_minor_inst_valid_fullsupport, decorated_func_full_support):
+        """Check that fully supported IATI Versions are detected as such."""
+        version = std_ver_minor_inst_valid_fullsupport
+
+        assert iati.version._is_fully_supported(version) is True
```

Comment:
Have put `is True` and `is False` to more thoroughly assert that a boolean value is returned, rather than something else that happens to evaluate to `True` or `False`.

Reply:
Makes sense.